Orthtree/include/CGAL/Orthtree.h

```diff
@@ -459,77 +395,162 @@ class Orthtree
-    compatible with the type of the orthree
+    \tparam Traversal a model of `OrthtreeTraversal`
 
-    \param traversal the instance of `Traversal` used
+    \param traversal class defining the traversal strategy
 
-    \return a forward input iterator over the nodes of the tree
+    \return a forward input iterator over the node indices of the tree
    */
-  template<typename Traversal>
-  Node_range traverse(const Traversal &traversal = Traversal()) const {
+  template <typename Traversal>
+  Node_index_range traverse(Traversal traversal) const {
 
-    Node first = traversal.first(m_root);
+    Node_index first = traversal.first_index();
 
-    Node_traversal_method_const next
-      = [&](const Node& n) -> Node { return traversal.next(n); };
+    auto next = [=](const Self&, Node_index index) -> Maybe_node_index {
+      return traversal.next_index(index);
+    };
 
-    return boost::make_iterator_range(Traversal_iterator<Node>(first, next),
-                                      Traversal_iterator<Node>());
+    return boost::make_iterator_range(Index_traversal_iterator<Self>(*this, first, next),
+                                      Index_traversal_iterator<Self>());
   }
 
+
   /*!
-    \brief constructs the bounding box of a node.
+    \brief convenience method for using a traversal without constructing it yourself
+
+    \tparam Traversal a model of `OrthtreeTraversal`
+
+    \param args Arguments to to pass to the traversal's constructor, excluding the first (always an orthtree reference)
 
-    \note The object constructed is not the bounding box of the point
-    subset inside the node, but the bounding box of the node itself
-    (cubic).
+    \return a forward input iterator over the node indices of the tree
    */
-  Bbox bbox(const Node &node) const {
-    // Determine the side length of this node
-    FT size = m_side_per_depth[node.depth()];
+  template <typename Traversal, typename ...Args>
+  Node_index_range traverse(Args&& ...args) const {
+    return traverse(Traversal{*this, std::forward<Args>(args)...});
+  }
 
-    // Determine the location this node should be split
-    Array min_corner;
-    Array max_corner;
-    for (int i = 0; i < Dimension::value; i++) {
+  /*!
+    \brief constructs the bounding box of a node.
+
+    \note The object constructed is not the bounding box of the node's contents,
+    but the bounding box of the node itself.
 
-      min_corner[i] = m_bbox_min[i] + (node.global_coordinates()[i] * size);
-      max_corner[i] = min_corner[i] + size;
+    \param n node to generate a bounding box for
+
+    \return the bounding box of the node n
+   */
+  Bbox bbox(Node_index n) const {
+    using Cartesian_coordinate = std::array<FT, Dimension::value>;
+    Cartesian_coordinate min_corner, max_corner;
+    std::size_t node_depth = depth(n);
+    Bbox_dimensions size = m_side_per_depth[node_depth];
+    const std::size_t last_coord = std::pow(2,node_depth)-1;
```

Comment:
```suggestion
    const std::size_t last_coord = ( 1 << node_depth)-1;
```
not only that `pow()` triggers a conversion warning, it is slower than shifting.